ooniprobe/AppDelegate.mm

```diff
@@ -143,6 +155,53 @@ -(void)application:(UIApplication *)application didReceiveLocalNotification:(UIL
     //TODO start your long running bg task here
 }
 
+- (BOOL)application:(UIApplication *)application handleOpenURL:(NSURL *)url {
+    //NSLog(@"url recieved: %@", url);
+    //NSLog(@"query string: %@", [url query]);
+    
+    //creating parameters dict
+    NSDictionary *dict = [DictionaryUtility parseQueryString:[url query]];
+    NSDictionary *parameters = [DictionaryUtility getParametersFromDict:dict];
+    
+    NSString *minimum_version = [parameters objectForKey:@"mv"];
+    if (minimum_version != nil){
+        if ([minimum_version compare:[[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleShortVersionString"] options:NSNumericSearch] == NSOrderedDescending) {
+            //actualVersion is lower than the requiredVersion
+            UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"ooniprobe_outdate", nil) message:NSLocalizedString(@"ooniprobe_outdate_msg", nil) delegate:nil cancelButtonTitle:NSLocalizedString(@"Cancel", nil) otherButtonTitles:NSLocalizedString(@"ok", nil), nil];
```

Comment:
@lorenzoPrimi I was wondering: can `alloc` fail on iOS such that `nil` is returned? What happens in `iOS` if you access a `nil` value? Segmentation fault or is this converted into an "exception" by the framework?

Reply:
if you try to access a parameter into a var that is nil it crashes the app

Reply:
okay, and can `alloc` fail?

Reply:
which alloc?

Reply:
`[UIAlertView alloc]`

Reply:
no idea. I don't even think it can happen such scenario :D

Reply:
I tried to dig in a little bit. What I have found out (but this is still cursory) is the following:

1. ["Objective-C safely handles sending messages to nil by ignoring them."](https://www.quora.com/Why-do-I-need-to-do-Class-*obj-Class-alloc-init-on-one-line/answer/Andy-Dent), which means that `Foo *foo = [[Foo alloc] init]` is safe

2. [that there are specific patterns that you should follow in the init handler](https://stackoverflow.com/a/10358995)

Reply:
So, in short: the scenario _can_ happen but you don't need to worry about it (at least in this context) because attempting to send messages to `nil` is something supported by ObjC.